### ISSUE TITLE
2.0.26 doesn't work on big-endian CPUs

### ISSUE THREAD

**Issue by sataevr** [DNSCrypt/dnscrypt-proxy#925]
Resolve process in 2.0.26 is very slowly. Dnsleaktest shows 1-3 servers while in .25 was 10-15+. dnsmasq and .toml configuration from Wiki and didn't change since .25.

**Comment by jedisct1** [comment]
Is that using DNSCrypt? DoH? Both?

**Comment by jedisct1** [comment]
What CPU is your router running?

**Comment by sataevr** [comment]
> Is that using DNSCrypt? DoH? Both?
> What CPU is your router running?

Both.
TP-Link TL-WDR3600 (Atheros AR9344), OpenWrt built from sources: "openwrt-18.06" branch v194.2-r7853-f6de1fa6c6. Uptime is 45min, load average from LuCI is 0.01, 0.02, 0.00.

**Comment by jedisct1** [comment]
I think I found the issue. It's not exactly related to OpenWRT, but rather the Mips CPU of these routers.

In version 2.0.26, some cryptographic operations have been switched to using a new library (CIRCL, by Cloudflare).

Unfortunately, that library doesn't work properly on big-endian CPUs.

I've switched back to the previous library and released version 2.0.27.

As a bonus, version 2.0.27 is a little bit smaller than 2.0.26.

Give it a try!
